### ISSUE TITLE
regression: parse error related to comment lines

### ISSUE THREAD

**Issue by irmen** [irmen/prog8#47]
Since merging #45 a new parse error popped up when using comment lines alternated with empty lines:

```
; what

; what

main {
    sub start() {
    }
}
```

> examples/test.p8:5:0: mismatched input 'main' expecting \<EOF\>

**Comment by meisl** [comment]
Ooh... that PR didn't hold too long. I'll amend it with a test for *that* and look into it.

**Comment by irmen** [comment]
no worries, here's a skeleton for the test that I worked on before going for an ice cream:

```
    @Test
    fun testCommentLinesWorkWithEmptyLines() {
        val srcText = """; comment before empty line

; comment after empty line

main {
    sub start() {
    }
}
"""
        val parseTree = parseModule(srcText)
        assertTrue(parseTree.block().size > 0)
    }
```

**Comment by meisl** [comment]
Well, that's the nice thing about automated tests ;)

Anyways, just pushed to #45.

The problem was this:
* comments go into their own channel, and aren't seen by the parser
* EOLs, however are, since they are used as end-of-statement marker etc
* By looking at just the EOL lexer rule, one can get the impression, that the parser will never see consecutive EOLs - it seems they should "eat up" all prior vertical whitespace
* But that's not true, putting comments in between EOLs has the parser see consecutive EOLs (which are separated by the invisible comments)

So the fix itself was straight-forward:
* `EOL?` -> `EOL*`
* `EOL` -> `EOL+`

**Comment by meisl** [comment]
Hm, since #45 was already merged the fix didn't appear there. So I just opened #48.

**Comment by meisl** [comment]
Btw: thanks for checking on me. It's great working with you 👍

**Comment by irmen** [comment]
fixed by merging #48 .

I appreciate your extensive problem reports and analyses they really help getting to the core of the matter

**Comment by meisl** [comment]
Thank you :D
